### PR TITLE
Some pipe2jpeg improvements

### PR DESCRIPTION
**Pull request by michaeljherrmann**
We've been using your library (thanks for making it!), but came across two cases we needed to improve. Thought I would push these upstream as well in case you would like to integrate it as well:

1. Smarter forward seeking when looking for end of input (EOI) after receiving the start of input (SOI). Before it was just a fixed offset, which could cause issues if the forward skip missed the next EOI. The new approach uses half of the size of the last received jpeg (so the assumption here is that the input frames won't be widely different sizes, which I think is reasonable).

2. I had an edge case where ffmpeg split the EOI across two chunks. So the last byte of the previous chunk had the first byte of EOI and the next chunk started with the second byte of EOI. I've updated the logic to handle this case, and also the related case of where the SOI could be split (but that case was not something I've come across yet).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "pipe2jpeg",
-  "version": "0.3.1",
+  "name": "lumen5-pipe2jpeg",
+  "version": "0.3.3",
```
Comment (michaeljherrmann) [inline package.json:3]:
```suggestion
  "version": "0.3.2",
```

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "pipe2jpeg",
-  "version": "0.3.1",
+  "name": "lumen5-pipe2jpeg",
```
Comment (michaeljherrmann) [inline package.json:2]:
```suggestion
  "name": "pipe2jpeg",
```

**Comment by kevinGodell** [comment]
As you may have noticed, I published a fix for the jpeg markers being missed when split across chunks. Had to create a bunch of tests to ensure that all edge cases are covered and that it performs efficiently.

I am currently adding a new feature that allows you to set readableObjectMode and bufferConcat to control the output. In the cases where you don't always need a re-assembled jpeg buffer, then **NOT** concatenating every jpeg can have a great improvement on performance since Buffer.concat can be very expensive if we are dealing with many chunks in the case of piping a very large resolution jpeg. The funny thing is that I accidentally stumbled across the performance issue of Buffer.concat when I was working on byte skipping. see the difference in times when running [tests/test10.js](https://github.com/kevinGodell/pipe2jpeg/blob/master/tests/test10.js#L35) vs [tests/test11.js](https://github.com/kevinGodell/pipe2jpeg/blob/master/tests/test11.js#L35)

As for the byte skipping feature, I have a idea and will be working on that next when time permits. My testing has shown that it has to include more than just using indexOf with an offset.

**Comment by kevinGodell** [comment]
The latest feature in 0.3.4 allows for dynamically setting the byteOffset value so you can adjust it externally using the getter/setter. This [test](https://github.com/kevinGodell/pipe2jpeg/blob/master/tests/test13.js#L41) shows how you can sample the 1st 10 jpegs and use the lengths to calculate the byteOffset. I felt that it had to be an externally controlled value instead of baking it inside. Beware of different byte size jpegs if streaming from a cctv cam and it changes between day and night mode.
